src/coreclr/jit/jitconfig.cpp

```diff
@@ -92,7 +79,7 @@ void JitConfigValues::MethodSet::destroy(ICorJitHost* host)
     }
     if (m_list != nullptr)
     {
-        host->freeMemory(static_cast<void*>(m_list));
+        host->freeMemory(const_cast<char*>(m_list));
```

Comment:
This `const_cast` should be factored out. It looks like this is all the result of the return of `getStringConfigValue` being marked as returning a `const char*`. It doesn't need to be that, especially if it is a local member. I would expose the field as a `const char*`, but internal to the type simply make it `char*` and avoid the very suspect `const_cast`.

Reply:
Actually, I think the issue here is this should be `host->freeStringConfigValue(m_list);`. I'm assuming this is a result of the initialization paths implemented via macro expansion.

Reply:
Yeah, good catch. I renamed a few things and added some documentation to make it more clear how the ownership works of this string.